### PR TITLE
Protocol updates and fixes

### PR DESCRIPTION
**Pull request by chrisdp**

### CODE REVIEW

File: src/debugProtocol/Debugger.ts [hunk 1]
```diff
@@ -281,96 +290,100 @@ export class Debugger {
...
-    private parseUnhandledData(unhandledData: Buffer): boolean {
+    private parseUnhandledData(buffer: Buffer): boolean {
         if (this.handshakeComplete) {
-            let debuggerRequestResponse = new Response(unhandledData);
-            if (debuggerRequestResponse.success) {
+            let debuggerRequestResponse = this.watchPacketLength ? new ProtocolEventV3(buffer) : new ProtocolEvent(buffer);
+            let packetLength = debuggerRequestResponse.packetLength;
+            let slicedBuffer = packetLength ? buffer.slice(4) : buffer;
 
+            console.log('incoming data', debuggerRequestResponse)
```
Comment (TwitchBronBron) [inline src/debugProtocol/Debugger.ts:299]: Use the logger for this, not `console.log`. ;)

File: src/debugProtocol/Debugger.ts [hunk 2]
```diff
@@ -281,96 +290,100 @@ export class Debugger {
...
+    private removedProcessedBytes(responseHandler: {requestId: number, readOffset: number}, unhandledData: Buffer, packetLength = 0) {
+        if (responseHandler.requestId > 0 && this.activeRequests[responseHandler.requestId]) {
             delete this.activeRequests[responseHandler.requestId];
         }
 
         this.emit('data', responseHandler);
 
-        this.unhandledData = unhandledData.slice(responseHandler.readOffset);
+        this.unhandledData = unhandledData.slice(packetLength ? packetLength : responseHandler.readOffset);
+        console.log(this.unhandledData.length, responseHandler);
```
Comment (TwitchBronBron) [inline src/debugProtocol/Debugger.ts:386]: Use the logger for this, not `console.log`

File: src/debugProtocol/Constants.ts [hunk 3]
```diff
@@ -14,65 +14,80 @@ export const enum COMMANDS {
...
 //#region RESPONSE CONSTS
-export const ERROR_CODES = {
-    0: 'OK',
-    1: 'OTHER_ERR',
-    2: 'UNDEFINED_COMMAND',
-    3: 'CANT_CONTINUE',
-    4: 'NOT_STOPPED',
-    5: 'INVALID_ARGS'
-};
+export enum ERROR_CODES {
```
Comment (TwitchBronBron) [inline src/debugProtocol/Constants.ts:25]: Let's rename all these enum names using PascalCase instead of SNAKE_CASE.

**Comment by chrisdp** [comment]
Closed as all logic was merged into #45

**Pull request closed without merging**